pyiron_contrib/atomistic/atomicrex/atomicrex_job.py

```diff
@@ -0,0 +1,127 @@
+    def to_hdf(self, hdf=None, group_name=None):
+        if hdf == None:
+            hdf = self.project_hdf5
+        self.input.to_hdf()
+        self.potential.to_hdf()
+        self.structures.to_hdf()
+        return
+    """
+
+    """
+    def from_hdf(self, hdf=None, group_name=None):
+        if hdf == None:
+            hdf = self.project_hdf5
+        self.input.from_hdf()
+        self.potential.from_hdf()
+        self.structures.from_hdf()
+        return
+    """
+
+    @property
+    def publication(self):
+        return {
+            "atomicrex": {
+                "atomicrex": {
+                    "title": "Atomicrex-a general purpose tool for the construction of atomic interaction models",
+                    "journal": "Modelling and Simulation in Materials Science and Engineering",
+                    "volume": "25",
+                    "number": "5",
+                    "year": "2017",
+                    "issn": "0965-0393",
+                    "doi": "10.1088/1361-651X/aa6ecf",
+                    "url": "https://doi.org/10.1088%2F1361-651x%2Faa6ecf",
+                    "author": ["Alexander Stukowski", "Erik Fransson", "Markus Mock", "Paul Erhart"],
+                }
+            }
+        }
+    
+    def collect_output(self, cwd=None):
+        #self.input.from_hdf(self._hdf5)
+        if cwd is None:
+            cwd = self.working_directory
+
+        filepath = f"{cwd}/atomicrex.out"
+        with open(filepath) as f:
+            lines = f.readlines()
+        l_index_params = None
+        l_index_complete = None
+        for i, line in enumerate(lines):
+            if line.startswith("ERROR"):
+                self.status.aborted=True
+                self.output.error = line
+                return
+            if line.startswith("Potential parameters"):
+                l_index_params = i
+            if line.startswith("Fitting process complete."):
+                l_index_complete = i
+                struct_lines = self.output._get_structure_lines(l_index_complete, lines)
+                parameter_lines = self.output._get_parameter_lines(l_index_params, lines)
+                self.structures._parse_final_properties(struct_lines)
+                self.potential._parse_final_parameters(parameter_lines)
+                break
+
+
+    def convergence_check(self):
+        return
+    
+    def write_input(self, directory=None):
+        if directory is None:
+            directory = self.working_directory
+        self.input._write_xml_file(directory = directory)
+        self.potential.write_xml_file(directory = directory)
+        self.structures.write_xml_file(directory = directory)
+    
+    """
+    def run(self):
+        cwd = self.working_directory
+        self._create_working_directory()
+        self.write_input(directory=cwd)
+        with open(f"{cwd}/atomicrex.out", "w") as f:
+            subprocess.run(["atomicrex", "main.xml"], stdout=f, stderr=f, text=True, cwd=cwd)
```

Comment:
We should package the executable in the resource folder somehow, but @jan-janssen knows more about this.

Reply:
I commented out the run function to use the default one and added a shell script in my ressources folder for atomicrex, so it now looks like this:
~/pyiron/resources/atomicrex/bin/run_atomicrex_0.1.6.sh which contrains
#!/bin/bash
exec atomicrex main.xml >> atomicrex.out;

This works, but it does not find the executable path using the default function used for this because it is one module level deeper than the other codes. To make it more clear the path is searched using something like self.__module__.split(".")[1] by default, but this would require self.__module__.split(".")[2] to work. Maybe I can just redefine the function in my job class (where I don't know if it will break something @jan-janssen ) or move atomicrex one level up, i.e. next to atomistic. I thin my first description on this was unclear sorry for that.